phalcon/db/adapter/pdo.zep

```diff
@@ -85,9 +85,12 @@ abstract class Pdo extends Adapter
 	 */
 	public function connect(descriptor = null)
 	{
-		var username, password, dsnParts, dsnAttributes,
+		var eventsManager,
+			username, password, dsnParts, dsnAttributes,
 			persistent, options, key, value;
 
+		let eventsManager = <ManagerInterface> this->_eventsManager;
```

Comment:
True but the behaviour and location is correct as far as just the firing of the event.  It also then sets up workarounds like the one Andres suggested such as extending the parent and child adapters to prevent the connect from occurring in the constructor.